pandas/core/generic.py

```diff
@@ -532,13 +532,38 @@ def pop(self, item):
 
         return result
 
-    def squeeze(self, **kwargs):
-        """Squeeze length 1 dimensions."""
-        nv.validate_squeeze(tuple(), kwargs)
+    def squeeze(self, axis=None):
+        """Squeeze length 1 dimensions.
+
+        Parameters
+        ----------
+        axis : None or int or tuple of ints, optional
```

Comment:
so you don't need virtually any of this validation, simply

``axis = self._get_axis_number(axis)``, this can only accept a single axis (or None by-default)